### PR TITLE
Sto refactor ssl topics doc 3628 v2

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request

This PR is a major refactor of the STO doc for setting up STO to run scans with custom SSL certificates. The following topics need an end-to-end review for technical accuracy and completeness:

- [Run STO scans with custom SSL certificates](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto)
- [Add custom certificates to a delegate](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate)
- [Add custom certificates to a pipeline](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines)
- [Troubleshoot SSL in STO](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-troubleshooting-in-sto)
- [Configure STO to Download Images from a Private Registry](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry)

### CODE REVIEW

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Harness supports the following setups for running scans with custom certificates: 
+
+- [Add your custom certificates to the delegate.](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md) This workflow is recommended if ALL of the following are true:
+
+  - You're using a Kubernetes or Docker delegate.
+
+  - You can configure the delegate directly.
+
+  - Your pipeline doesn't require any additional files, such as license or context files, to run scans.
```
Comment (r0610205) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto.md:17]: I don't think we need 3rd condition here

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+- You're using any delegate type other than Kubernetes or Docker, such as a Harness Cloud delegate.
+
+- You cannot configure the delegate directly. 
+
+- The external scanner requires additional files, such as context or license files, to run scans. 
+
+## Important notes
+
+- This workflow is supported for Kubernetes and Docker delegates only, and requires direct access to the delegate. If you don't meet these requirements, you can add your certs to [individual pipelines](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines) or to [custom scanner images](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/use-private-registry-for-sto-scanner-images) instead.
```
Comment (r0610205) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md:31]: Repeating all 3 workflows here is confusing, we already talking about delegate here. If you want to have reference just add a link to [ssl-setup-in-sto.md](https://github.com/harness/developer-hub/pull/3068/files#diff-a615536b5d447ee52a5ed1124f35f0e5915d63ea62ef8f22981378208caaf68b)

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto.md (hunk 1 above)
Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto.md]: Updated here: [Supported workflows for adding custom SSL certificates](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto#supported-workflows-for-adding-custom-ssl-certificates)

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  
+- [Add your custom certificates to individual pipelines.](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines.md) This workflow is recommended if ANY of the following are true:
+
+  - You're using any delegate type other than Kubernetes or Docker, such as a Harness Cloud delegate.
+
+  - You cannot access or customize your delegate directly. 
+
+  - The external scanner requires additional files, such as context or license files, to run scans. For example, ZAP scans might require files as auth scripts.
```
[the remark this answers is not in the document]
Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto.md]: Updated here: [Supported workflows for adding custom SSL certificates](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto#supported-workflows-for-adding-custom-ssl-certificates)

I rewrote this as 

[Add your custom certificates to individual pipelines.](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines) This workflow is recommended if either of the following are true:

- You're using any delegate type other than Kubernetes or Docker, such as a Harness Cloud delegate.

- You cannot access or customize your delegate directly.

  You can also use this workflow if the external scanner requires additional files, such as auth script or license files, to run scans.

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines.md [hunk 4]
```diff
@@ -1,106 +1,59 @@
...
-- To troubleshoot SSL issues, go to [Troubleshooting tips](#troubleshooting-tips) below. 
+- You're using any delegate type other than Kubernetes or Docker, such as a Harness Cloud delegate.
 
-:::
+- You cannot access or customize your delegate directly. 
 
+- The external scanner requires additional files, such as auth script or license files, to run scans. For example, ZAP scans might require context files as [noted](#important-notes) below.
 
-## Adding certificates to a Kubernetes delegate
+The [delegate workflow](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md) is recommended if ALL of the following are true:
```
[the remark this answers is not in the document]
Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines.md]: Updated here: [When to use this workflow](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines#when-to-use-this-workflow)

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md (hunk 2 above)
Comment (douglas-j-bothwell) [reply docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md]: Updated here: [When to use this workflow](https://64ed3bac89303513a9e4b007--harness-developer.netlify.app/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate#when-to-use-this-workflow)

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Important notes
+
+- This workflow is supported for Kubernetes and Docker delegates only, and requires direct access to the delegate. If you don't meet these requirements, you can add your certs to [individual pipelines](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines) or to [custom scanner images](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry) instead.
+
+- You must have **root access** to perform the [workflows](#workflow-descriptions) referenced below.
+
+- Make sure that your certificates meet all requirements of the external scan tool. Your certificates must be valid, unexpired, and have a complete trust chain. 
+
+- STO supports certificates in PEM and DER (Distinguished Encoding Rules) format.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-delegate.md:27]:
```suggestion
- STO supports certificates in PEM and Distinguished Encoding Rules (DER) format.
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines.md [hunk 6]
```diff
@@ -1,106 +1,49 @@
...
-          secret:  
-            secretName: addcerts  
-            items:  
-            - key: ca.bundle  
-              path: ca.bundle
-```
+- Make sure that your certificates meet all requirements of the external scan tool. Your certificates must be valid, unexpired, and have a complete trust chain. 
 
-</details>
+- STO supports certificates in PEM and DER (Distinguished Encoding Rules) format.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/add-certs-to-pipelines.md:27]:
```suggestion
- STO supports certificates in PEM and Distinguished Encoding Rules (DER) format.
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 7]
```diff
@@ -10,7 +10,9 @@ This topic describes how to override the default behavior and use a private regi
 
 ### Create STO scanner images with your own SSL certificates (_optional_)
 
-You can set up your STO scan images and pipelines to run scans as non-root and establish trust for your own proxies using self-signed certificates. This workflow supports any STO-compatible scanner that can run natively without root access. This workflow also supports build environments that use a self-signed proxy server between the Harness Delegate and the Harness Manager.
+Harness STO supports [three workflows](/docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/add-custom-certs/ssl-setup-in-sto#supported-workflows-for-adding-custom-ssl-certificates) for running scans with custom certificates.
+
+In this workflow, you set up your STO scan images and pipelines to run scans as non-root and establish trust for your own proxies using self-signed certificates. This workflow supports any STO-compatible scanner that can run natively without root access. This workflow also supports build environments that use a self-signed proxy server between the Harness Delegate and the Harness Manager.
```
Comment (brian-f-harness) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:15]:
```suggestion
In this workflow, you set up your STO scan images and pipelines to run scans as non-root and establish trust for your own proxies using self-signed certificates. This workflow supports any STO-compatible scanner that can run natively without root access. This workflow also supports build environments that use a self-signed proxy server between the Harness Delegate and Harness Manager.
```

**Review by brian-f-harness — APPROVED** [review]
LGTM. Left a couple minor suggestions.

**Pull request merged**